### ISSUE TITLE
It is possible to create xdr_vector<N> bigger than N

### ISSUE THREAD

**Issue by vogel** [xdrpp/xdrpp#11]
Simple example from stellar-core:

Signature out(200, 0);

Signature has limit of 64 but it is not checked in this constructor, only on resize/append

Seems like xstring solves it by its own constructors:

  template<typename...Args> xstring(Args&&...args)
    : string(std::forward<Args>(args)...) { validate(); }

**Comment by stanford-scs** [comment]
This is correct that xvector constructor does not validate.  In general, the types are paranoid on (un)marshaling, but do not check every single thing that could increase their size.  Strings seem particularly easy to overflow, and thus worth checking early even before marshaling.  I'm open to doing something similar to xvector if it seems helpful.
